gateway/api/authentication.py

```diff
@@ -88,6 +90,23 @@ def authenticate(self, request):
                         except User.DoesNotExist:
                             user = User(username=user_id)
                             user.save()
+                    if user_id is None:
+                        logger.warning("No user id.")
+
+                    if not verified:
+                        logger.warning("User is not verified.")
+
+                if verification_data is None:
```

Comment:
Is there any reason using **if xxxx** instead of **else** here?

Reply:
I found it hard to determine which else matched up with which if (one of the few times I wished python used brackets instead of whitespace), so I tried to make the blocks more explicit by using ifs

Reply:
I probably like `else` with comments.  It makes clear that all cases (only 2 though) are covered.